program/src/instructions/legacy/buy_spl.rs

```diff
@@ -222,6 +224,7 @@ impl<'info> BuyLegacySpl<'info> {
     fn transfer_currency(&self, to: &AccountInfo<'info>, amount: u64) -> Result<()> {
         transfer_checked(
             CpiContext::new(
+                // TODO: can this ever go wrong if a wrong program is called?
```

Comment:
Token account constraints for currency accounts check against the currency token program and the specific currency. The currency must match the one stored on the listing state. As long as those match it should be a valid buy_spl.